### PR TITLE
feat(frontend): add caldav arabic translation

### PR DESCRIPTION
**Pull request by knodelle**
Issue: #5492

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
  - The failing end-to-end test is not linked to this pr, so I will merge this as it is
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Comment by knodelle** [comment]
Some translations are missing, they will be added before this pr is closed

**Review by Haskaris — APPROVED** [review]

File: frontend/dist/assets/_caldavzap/localization.js [hunk 1]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+	pholderAlarmDateTODO: 'تاريخ التنبيه',
+	pholderAlarmTimeTODO: 'ساعة التنبيه',
+
+	pholderNoteTODO: 'ملاحظة',
+	txtNoteTODO: 'ملاحظة',
+
+	todoPrevNav: 'Show previous instance',
+	todoNextNav: 'Show next instance',
+	todoUncompletedPrevNav: 'Show previous uncompleted instance',
+	todoUncompletedNextNav: 'Show next uncompleted instance',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1250]:
```suggestion
	todoPrevNav: 'عرض المنصّة السابقة',
	todoNextNav: 'عرض المنصّة الموالية',
	todoUncompletedPrevNav: 'عرض المنصّة السابقة غير المكتملة',
	todoUncompletedNextNav: 'عرض المنصّة الموالية غير المكتملة',
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 2]
```diff
@@ -237,6 +254,41 @@ localization['zh_CN']=	/* thanks Fandy */
...
+	errHttpCommon: "خطأ: /'حفظ الاعدادات غير ممكن/' رمز الخطأ %%",
+	errHttp401: 'غير مسموح به',
+	errHttp403: 'ممنوع',
+	errHttp405: 'طريقة غير مسموح بها',
+	errHttp408: 'انتهاء مدة الطلب',
+	errHttp412: 'أحدهم قام بتعديله على الخادم',
+	errHttp500: 'خطأ داخلي في الخادم',
+	errHttp501: 'غير مطبق',
+
+	unsupportedBrowser: 'ملاحظة : لا يمكن تحمل المتصفح الذي تستعمله',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:288]:
```suggestion
	unsupportedBrowser: 'ملاحظة: لا يمكن تحمل المتصفح الذي تستخدمه!',
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 3]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+	txtNotChooseTodo: 'Error: Have to choose calendar!',
+	txtNotChoose: 'Error: Have to choose calendar!',
+
+	txtErrorDates: 'خطأ: تاريخ البداية يجب أن يسبق تاريخ النهاية',
+	txtErrorDatesTodo: 'Error: Start date must precede the end date!',
+
+	txtErorInputTodo: 'التسجيل غير ممكن : الرجاء تصحيح القيم غير الصحيحة',
+	txtDateTimeErrorTodo: 'يجب اختيار تاريخ و ساعة!',
+
+	txtErorInput: "خطأ : /التسجيل غير ممكن/' الرجاء تصحيح القيم المبرزة",
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1355]:
```suggestion
	txtAllDeletedTodo: 'نجاح: تم حذف المهمة!',
	txtAllDeleted: 'نجاح: تم حذف المناسبة!',

	txtNotChooseTodo: 'خطأ: يجب اختيار رزنامة!',
	txtNotChoose: 'خطأ: يجب اختيار رزنامة!',

	txtErrorDates: 'خطأ: تاريخ البداية يجب أن يسبق تاريخ النهاية!',
	txtErrorDatesTodo: 'خطأ: تاريخ البداية يجب أن يسبق تاريخ النهاية!',

	txtErorInputTodo: 'التسجيل غير ممكن: الرجاء تصحيح القيم غير الصحيحة!',
	txtDateTimeErrorTodo: 'يجب اختيار تاريخ و ساعة!',

	txtErorInput: "خطأ: /التسجيل غير ممكن'\ : الرجاء تصحيح القيم المبرزة!",
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 4]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+
+	txtStatusFiltersHeaderTODO: 'Filters',
+	txtStatusFiltersFooterTODO: 'أو بعده"%date%""ينتهي يوم',
+	txtStatusNeedsActionTODO: 'بانتظار إجراء',
+	txtStatusCompletedTODO: 'منتهي',
+	txtStatusInProcessTODO: 'جاري',
+	txtStatusCancelledTODO: 'ملغى',
+	txtPercentCompletedTODO: 'منتهي',
+
+	txtCalendarTODO: 'Calendar',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1223]:
```suggestion
	txtCalendarTODO: 'رزنامة',
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 5]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+	pholderLocation: 'المكان',
+
+	txtAllDay: 'كامل اليوم',
+	from: 'من',
+	to: 'الى',
+
+	pholderDateFrom: 'Date from',
+	pholderTimeFrom: 'Time from',
+	pholderDateTo: 'Date to',
+	pholderTimeTo: 'Time to',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1272]:
```suggestion
	pholderDateFrom: 'تاريخ البداية',
	pholderTimeFrom: 'ساعة البداية',
	pholderDateTo: 'تاريخ النهاية',
	pholderTimeTo: 'ساعة النهاية',
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/dist/assets/_caldavzap/localization.js [hunk 6]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+localizationCalDAV['ar']=	/* thanks Myriam Rabhi */
+{
+	_default_title_format_month_: 'yyyy MMM',
+	_default_title_format_week_: 'yyyy MMM d{ \'–\'[ yyyy][ MMM] d}',
+	_default_title_format_day_: 'yyyy MMM d, ddd',
+	_default_title_format_table_: 'yyyy MMM, ddd',
+	_default_column_format_agenda_: 'ddd, MMM d',
+
+	txtCalendar: 'رزنامة',
+	txtEnabledCalendars: 'Enabled Calendars',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1137]:
```suggestion
	txtEnabledCalendars: 'تفعيل الرزنامات',
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 7]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+	txtRepeatCustomMonthFifth: 'خامس',
+	txtRepeatCustomMonthLast: 'آخر',
+	txtRepeatCustomMonthCustom: 'مخصص',
+	txtRepeatCustomMonthMonth: 'اليوم من الشهر',
+	txtRepeatCustomYearLabel1: 'كل',
+	txtRepeatCustomYearLabel2: 'من',
+
+	txtRepeatEnd: 'نهاية',
+	txtEventCalendar: 'Calendar',
+	txtSelectCalendar: 'Select a calendar',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1312]:
```suggestion
	txtEventCalendar: 'رزنامة',
	txtSelectCalendar: 'اختيار رزنامة',
```

File: frontend/dist/assets/_caldavzap/localization.js [hunk 8]
```diff
@@ -1054,6 +1125,252 @@ $.extend(true, localization, localizationSharedCalDAV);
...
+	txtAlertSecondsAfter: 'ثواني بعد',
+
+	pholderAfterBeforeVal: 'قيمة',
+	pholderAlarmDate: 'تاريخ التنبيه',
+	pholderAlarmTime: 'ساعة التنبيه',
+
+	pholderNote: 'ملاحظة',
+
+	eventPrevNav: 'Show previous instance',
+	eventNextNav: 'Show next instance',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:1335]:
```suggestion
	eventPrevNav: 'عرض المنصّة السابقة',
	eventNextNav: 'عرض المنصّة الموالية',
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/dist/assets/_caldavzap/localization.js [hunk 9]
```diff
@@ -237,6 +254,41 @@ localization['zh_CN']=	/* thanks Fandy */
 };
 var localizationShared = new Object();
 
+localizationShared['ar']=	/* Thanks Myriam Rabhi */
+{
+	_name_: 'العربية',
+	_default_datepicker_format_: 'يوم/شهر/سنة',
```
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:260]: dd/mm/yyyy

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/dist/assets/_caldavzap/localization.js (hunk 9 above)
Comment (gbonaspetti) [inline frontend/dist/assets/_caldavzap/localization.js:260]:
```suggestion
	_default_datepicker_format_: 'dd/mm/yyyy',
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**